hibernate-core/src/test/java/org/hibernate/query/hhh13670/HHH13670Test.java

```diff
@@ -101,6 +100,52 @@ public void testSubTypeJoinWithTableGroupJoins() {
         });
     }
 
+    @Test
+    public void testSubTypePropertyReferencedFromEntityJoinInSyntheticSubquery() {
+        doInJPA(this::sessionFactory, em -> {
+            List<Tuple> resultList = em.createQuery(
+                    "SELECT  subB_0.id, subA_0.id, subB_0.id, subA_0.id FROM SubB subB_0 INNER JOIN SubA subA_0 ON 1=1 WHERE (EXISTS (SELECT 1 FROM subB_0.parent _synth_subquery_0 WHERE subA_0.id = _synth_subquery_0.id)) ORDER BY subB_0.id ASC, subA_0.id ASC", Tuple.class)
+                    .getResultList();
+
+            assertEquals(1, resultList.size());
+        });
+    }
+
+    @Test
+    public void testSubTypePropertyReferencedFromEntityJoinInSyntheticSubquery2() {
+        doInJPA(this::sessionFactory, em -> {
+            List<Tuple> resultList = em.createQuery(
+                    "SELECT  subB_0.id, subA_0.id, subB_0.id, subA_0.id FROM SubB subB_0 INNER JOIN SubA subA_0 ON 1=1 WHERE (EXISTS (SELECT 1 FROM Super s WHERE subA_0.id = s.parent.id)) ORDER BY subB_0.id ASC, subA_0.id ASC", Tuple.class)
```

Comment:
I think what I actually meant was `SELECT  subB_0.id, subA_0.id, subB_0.id, subA_0.id FROM SubB subB_0 INNER JOIN SubA subA_0 ON 1=1 WHERE (EXISTS (SELECT 1 FROM Super s WHERE s.id = subA_0.parent.id)) ORDER BY subB_0.id ASC, subA_0.id ASC` to test if an expression referring to a superclass property appearing just in the subquery WHERE clause also triggers the addition of the superclass table

Reply:
I have ran that query as well and it returns the expected empty row set.

I've also ran `SELECT subB_0.id, subA_0.id, subB_0.id, subA_0.id FROM SubB subB_0 INNER JOIN SubA subA_0 ON 1=1 WHERE (EXISTS (SELECT 1 FROM Super s WHERE s.id = subB_0.parent.id)) ORDER BY subB_0.id ASC, subA_0.id ASC` , which correctly returns 6 rows.